### DISCUSSION TITLE
Parent Theme CSS structure

### DISCUSSION THREAD

**Discussion by danielmrey** [holger1411/understrap#1370] [Ideas]
Hi:

I am using my UnderStrap Child Theme for several clients. So when there is an update of the Parent Theme, it would be great if we didn't have to touch the Child Theme unless it is necessary, that is why I use the Parent Theme CSS in my Child.

I have notice that the Parent Theme has got 3 main areas of rules:
1. BootStrap + Contact Form 7
2. UnderStrap + Galleries + WooCommerce
3. Font Awesome + Contact Form 7

Wouldn't it be better to split the Parent Theme CSS in 3 files?:
1. bootstrap.css
2. understap.css (UnderStrap + Galleries + WooCommerce + Contact Form 7)
3. font-awesome.css

So we can replace or activate/deactivate in the Child Theme files as needed?... Specially if Font Awesome is not needed.

Thank you.

Daniel.

**Comment by bacoords** [comment]
This is a common topic of consideration.  Either method has pros/cons. There's a few things to consider here, because when you build a child theme, you typically want total control over assets and versions.

For example, you might want to load a smaller version of Bootstrap, bundle an additional dependency, or keep stylesheets locked at a certain version. This lets you build on top of the framework without worry about future updates breaking your code. 

Or maybe you want to override default Bootstrap variables to create your own theme preferences. Rather than manually overriding bootstrap in a separate stylesheet, you're compiling the sass automatically, etc. 

The theme was designed to concat/compile all assets together and leave you with ONE stylesheet and ONE javascript file, typically for pagespeed benefits (which is another question). Then, if you want to activate/deactivate in a child theme, you can just comment them out in the SASS files before you build the final stylesheet.

Definitely open to hearing more thoughts/opinions on this.

**Comment by danemorgan** [comment]
I would say that the approach we already follow is the way to go. We should be bundling the core that is used by the 80% and for the 20% cases, the whole point is that individual devs can add or remove modules and packages to the SASS.

I would support migrating from Font Awesome to Bootstrap icons. Even to the point of backporting them to Bootstrap4. We're a Bootstrap-based theme and Font Awesome is an overly heavy weight to carry for 99% of use cases.
